File: robots/java/RobotTest.java
import junit.framework.*;

public class RobotTest extends TestCase{
    public RobotTest(String name) {super(name);}

    public void testRobot() {
        Machine sorter = new Machine("Sorter", "left");
        sorter.put("chips");
        Machine oven = new Machine("Oven", "middle");
        Robot robot = new Robot();

        assertEquals("chips", sorter.bin());
        assertNull(oven.bin());
        assertNull(robot.location());
        assertNull(robot.bin());

        robot.moveTo(sorter);
        robot.pick();
        robot.moveTo(oven);
        robot.release();

        assertNull(robot.bin());
        assertEquals(oven, robot.location());
        assertNull(sorter.bin());
        assertEquals("chips", oven.bin());
    }
}

File: robots/vb/RobotTest.vb
Imports NUnit.Framework

Namespace ReportGeneratorVB.Tests
    <TestFixture()> _
    Public Class RobotTest
        <Test()> _
        Public Sub TestRobot()
            Dim sorter As New Machine("Sorter", "left")
            sorter.Put("chips")
            Dim oven As New Machine("Oven", "middle")
            Dim robot As New Robot()

            Assert.That("chips", [Is].EqualTo(sorter.Bin()))
            Assert.That(oven.Bin(), [Is].Null)
            Assert.That(robot.Location(), [Is].Null)
            Assert.That(robot.Bin(), [Is].Null)

            robot.MoveTo(sorter)
            robot.Pick()
            robot.MoveTo(oven)
            robot.Release()

            Assert.That(robot.Bin(), [Is].Null)
            Assert.That(oven, [Is].EqualTo(robot.Location()))
            Assert.That(sorter.Bin(), [Is].Null)
            Assert.That("chips", [Is].EqualTo(oven.Bin()))
        End Sub
    End Class
End Namespace